query/physicalplan/aggregate.go

```diff
@@ -658,16 +646,20 @@ func (a *CountAggregation) Aggregate(pool memory.Allocator, arrs []arrow.Array)
 // runAggregation is a helper to run the given aggregation function given
 // the set of values. It is aware of the final stage and chooses the aggregation
 // function appropriately.
-func runAggregation(
-	finalStage bool,
-	fn AggregationFunction,
-	pool memory.Allocator,
-	arrs []arrow.Array,
-) (arrow.Array, error) {
-	if _, ok := fn.(*CountAggregation); ok && finalStage {
+func runAggregation(finalStage bool, fn logicalplan.AggFunc, pool memory.Allocator, arrs []arrow.Array) (arrow.Array, error) {
+	if len(arrs) == 0 {
+		return array.NewInt64Builder(pool).NewArray(), nil
+	}
+
+	aggFunc, err := chooseAggregationFunction(fn, arrs[0].DataType())
```

Comment:
Could you run the aggregation benchmarks? I'm wondering if doing this logical to physical agg func conversion each time we run an aggregation is detrimental to performance.

Reply:
```
thor@thors-MacBook-Pro frostdb % benchstat before.txt after.txt
name                  old time/op    new time/op    delta
Aggregation/sum-10      1.45ms ± 1%    1.43ms ± 1%  -0.73%  (p=0.003 n=9+9)
Aggregation/count-10    1.45ms ± 1%    1.43ms ± 1%  -1.36%  (p=0.000 n=10+9)
Aggregation/max-10      1.45ms ± 1%    1.44ms ± 1%  -0.99%  (p=0.003 n=10+10)

name                  old alloc/op   new alloc/op   delta
Aggregation/sum-10      1.40MB ± 0%    1.39MB ± 0%  -0.82%  (p=0.000 n=7+10)
Aggregation/count-10    1.40MB ± 0%    1.39MB ± 0%  -0.82%  (p=0.000 n=10+10)
Aggregation/max-10      1.40MB ± 0%    1.39MB ± 0%  -0.82%  (p=0.000 n=10+10)

name                  old allocs/op  new allocs/op  delta
Aggregation/sum-10       21.0k ± 0%     20.9k ± 0%  -0.69%  (p=0.000 n=10+10)
Aggregation/count-10     21.0k ± 0%     20.9k ± 0%  -0.69%  (p=0.000 n=10+10)
Aggregation/max-10       21.0k ± 0%     20.9k ± 0%  -0.69%  (p=0.000 n=10+10)
```

Reply:
Looks like it's negligible